biz.aQute.bndlib/src/aQute/bnd/build/Workspace.java

```diff
@@ -292,24 +292,52 @@ public Workspace(File workspaceDir) throws Exception {
 	}
 
 	public Workspace(File workspaceDir, String bndDir) throws Exception {
-		super(getDefaults());
+		super(new Processor(getDefaults()));
 		this.maven = new Maven(Processor.getExecutor(), this);
 		this.layout = WorkspaceLayout.BND;
 		workspaceDir = workspaceDir.getAbsoluteFile();
 		setBase(workspaceDir); // setBase before call to setFileSystem
 		addBasicPlugin(new LoggingProgressPlugin());
 		setFileSystem(workspaceDir, bndDir);
+
+		// we must process version defaults after the
+		// normal properties are read
+
+		fixupVersionDefaults(this.getParent()
+			.getProperties());
+
 		projects = new ProjectTracker(this);
 	}
 
 	private Workspace(WorkspaceLayout layout) throws Exception {
-		super(getDefaults());
+		super(new Processor(getDefaults()));
 		this.maven = new Maven(Processor.getExecutor(), this);
 		this.layout = layout;
 		setBuildDir(IO.getFile(BND_DEFAULT_WS, CNFDIR));
+		// we must process version defaults after the
+		// normal properties are read
+
+		fixupVersionDefaults(this.getParent()
+			.getProperties());
 		projects = new ProjectTracker(this);
 	}
 
+	private void fixupVersionDefaults(Properties props) throws IOException {
+		Version current = About.CURRENT.getWithoutQualifier();
+		String version = Strings.trim(getProperty(Constants.VERSIONDEFAULTS, current.toString()));
```

Comment:
I looked at how you handle the version in About and you do not use micro releases so far. I agree with that since micro releases should be bug fixes and not change public API. The defaults are public API since people can depend on it. So we can fix a bug in the defaults but we should not make any modification anyone can depend on. I.e. no new defaults.

Reply:
You would need to look in a micro release tag to see that we set Current to a micro version. So that would need to be handled here.

https://github.com/bndtools/bnd/commit/34d22ae2e42f1bf3fa7caa94a54ba4dec4762aa5#diff-2989e02189b9389c9d08724df134efb45be4d59982b8e0954759aa61e239dacdR54

Reply:
I had looked in About and they had no micro but I now realize this happens in the release branch :-)

I fix this.